### DISCUSSION TITLE
Tilt Position after complete close

### DISCUSSION THREAD

**Discussion by RaidObServer** [arendst/Tasmota#16435] [Q&A]
Hello Team,

i have a Shelly 2.5 with Tasmota 12.1 and use the venetian blind control for my shutter.
In the configuration guide the following is described for completely closed or open (tilt position):

If the shutter is moved from one position to another position the tilt will be restored AFTER the movement. If the shutter is fully opened or fully closed the tilt will be resetted. This means there is no tilt restore at the endpoints.

My question is, it is possible to restore the tilt position after the shutter is complete close (endpoint) or can i add this to my settings?

best regards and many thanks
Stefan

**Comment by stefanbode** [comment]
The only workaround is to use 1 and 99% as endpositions. So far nobody want to restore the tilt, when totally open or close. On open it even makes no sense, but also on close a restore of the tilt is very unlikely. you can use shutterbutton commands to set close to 99 instead of 100% as an easy workaround

**Reply by RaidObServer** [reply]
Thank you stefanbode for your replay.
I'm totally agree with you for the open position.
From my point of view,  the closed position would be already useful that this can be set, because I usually want to close the shutter to 100% but the tilt should be to the set value.

For now i can use the workaround, but maybe this feature can be implemented in a future version.

Thank you and best regards,
Stefan

**Comment by matejdro** [comment]
It would be nice to have that feature working. For me personally, the use case is that I want to be able to tell my blinds to go to "position X and tilt Y" at once.

That means that if I want blinds to fully close and then tilt to open, I send "ShutterPosition 0", "ShutterTilt 90" commands in quick succession. Expected outcome after triggering those two commands is that blinds would go down and then tilt would fully open. But since tilt is not restored at the bottom, second command is essentially ignored.

**Comment by lalo-uy** [comment]
Can use Backlog to chain both commndas

**Reply by matejdro** [reply]
Thanks, but that represents a different issue: since tilt command would be executed after position command, it would restore previous position before setting tilt, right? Resulting in potentially redundant movement.

**Reply by matejdro** [reply]
Backlog also does not appear to work at the lowest position:

When running `Backlog ShutterPosition 0; ShutterTilt 90`,  Tasmota will ignore the second command. `Backlog ShutterPosition 1; ShutterTilt 90` works fine, though, so it appears to be related to the lack of  tilt restoration at the bottom.

**Reply by stefanbode** [reply]
Your finding is correct. Shutteropen and shutterclose (0,100) do behave different on the TILT. They do not restore the tilt to the value set but keep it where it is when open or close. Your workaround to use 1 or 99 as a position is absolutely the correct workaround for it.

**Reply by matejdro** [reply]
While this workaround works, it's a bit hacky and I feel it should not be necessary. Maybe there could be an option to enable/disable this behavior?

**Reply by stefanbode** [reply]
Yep. The reason behind is that whenever you open (for sure) and when close you do not want to maintain the tilt. This is also for people just using buttons to open or close. I'm just thinking if I can disable the behavior if there is a tilt command set after the shutter is moving. In this case the user intention is clear. This must also work with the backlog.

**Reply by stefanbode** [reply]
I submitted a new version adressing your need. https://github.com/arendst/Tasmota/pull/17186
As soon as it is merged into development you can test. The new behavior is:
If doing shutteropen, shutterclose, shutterposition 0 100 alls the same. Tilt will NOT recover.
New function: With backlog command you can set the tilt after move started. Tilt will be restored at the endposition.

**Reply by matejdro** [reply]
So, if I understand properly, running `Backlog ShutterPosition 0; ShutterTilt 90` will work now, but tilt will not recover in all other cases (when at bottom or top)?

**Reply by stefanbode** [reply]
Yes shutteropen and close same as 0 and 100 will stop after hitting the endpoint without any tilt. Here I have to ensure not breaking existing functionality.

**Reply by matejdro** [reply]
Sounds good, thanks a lot for your help so far. Can't wait to test the development build.

**Reply by matejdro** [reply]
Sorry for hijacking the thread, since you seem to be knowledgeable with the shutter code, do you have any idea what could cause this? https://github.com/arendst/Tasmota/discussions/17191

**Reply by matejdro** [reply]
Can confirm, works with latest development version. Thanks!

**Reply by matejdro** [reply]
@stefanbode It looks like new development version has a bug. Running the same shutter command will still actuate the blinds a bit.

For example:

1. Run `Backlog ShutterPosition 100;ShutterTilt 90`
2. Wait for blinds to finish moving
3. Run `Backlog ShutterPosition 100;ShutterTilt 90` again
4. Even though blinds are already at the given position and tilt, I can hear relay triggering momentarily

Here are logs of one such command being ran twice:

```
08:08:33.787 CMD: Backlog ShutterPosition 100;ShutterTilt 90
08:08:33.789 SRC: WebConsole from 192.168.10.2
08:08:33.792 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 34, Pld -99, Data 'ShutterPosition 100;ShutterTilt 90'
08:08:33.813 SRC: Backlog
08:08:33.816 CMD: Grp 0, Cmd 'SHUTTERPOSITION', Idx 1, Len 3, Pld 100, Data '100'
08:08:33.820 SHT: Pos. in: payload 100 (3), payload 100, idx 1, src 9
08:08:33.822 SHT: lastsource 9:, real 580000, target 580000, tiltreal: 84, tilttarget: 0, payload 100
08:08:33.824 SHT: Delay Start? var1 <99>=<>, max10s?
08:08:33.826 SHT: VenetianDelay: 23, Pos: 84, Dir: -1, Delta: 90, Dur: 25, StartP: 84, TgtP: 0
08:08:33.828 SRC: Shutter
08:08:33.830 SHT: Switched relay 2 by Shutter
08:08:33.835 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"POWER2":"ON"}
08:08:33.838 MQT: stat/tasmota_blinds_dnevna_desno/POWER2 = ON
08:08:33.941 SRC: Shutter
08:08:33.943 SHT: Switched relay 1 by Shutter
08:08:33.952 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"POWER1":"ON"}
08:08:33.956 MQT: stat/tasmota_blinds_dnevna_desno/POWER1 = ON
08:08:33.959 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Position1":100}
08:08:33.966 SHT: Shtr1 Real 580000, Start 580000, Stop 580000, Dir -1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 84
08:08:33.971 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Shutter1":{"Position":100,"Direction":-1,"Target":100,"Tilt":84}}
08:08:33.974 SHT: Time 0(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 580000, nStop 0, Trgt 580000, mVelo 0, Dir -1, Tilt 84, TrgtTilt: 0, Tiltmove: 1
08:08:33.979 SHT: Shtr1 Real 580000, Start 580000, Stop 580000, Dir -1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 84
08:08:33.985 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Shutter1":{"Position":100,"Direction":-1,"Target":100,"Tilt":84}}
08:08:34.044 SRC: Backlog
08:08:34.048 CMD: Grp 0, Cmd 'SHUTTERTILT', Idx 1, Len 2, Pld 90, Data '90'
08:08:34.051 SHT: TiltTarget 90, payload -99
08:08:34.053 SHT: Pos. in: payload  (2), payload -99, idx 1, src 23
08:08:34.056 SHT: Shtr1 Real 580000, Start 580000, Stop 580000, Dir -1, Delay 0, Rtc 0.05 [s], Freq 0, PWM 0, Tilt 84
08:08:34.062 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Shutter1":{"Position":100,"Direction":-1,"Target":100,"Tilt":84}}
08:08:34.068 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Position1":100}
08:08:34.073 SHT: Time 2(1), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 580000, nStop 0, Trgt 580000, mVelo 0, Dir -1, Tilt 81, TrgtTilt: 90, Tiltmove: 1
08:08:34.075 SHT: Stop 1 Mode 0 time 2
08:08:34.077 SRC: Shutter
08:08:34.079 SHT: Switched relay 1 by Shutter
08:08:34.084 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"POWER1":"OFF"}
08:08:34.089 MQT: stat/tasmota_blinds_dnevna_desno/POWER1 = OFF
08:08:34.092 SRC: Shutter
08:08:34.094 SHT: Switched relay 2 by Shutter
08:08:34.100 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"POWER2":"OFF"}
08:08:34.105 MQT: stat/tasmota_blinds_dnevna_desno/POWER2 = OFF
08:08:34.612 ADE: ACCMODE 0x2D2800, VRMS 6327344, 6326484, Period 4475, 4475, IRMS 1773, 1773, WATT 0, 77, VA 29, 139, VAR 0, 19
08:08:34.615 SHT: Shtr1 Real 580000, Start 580000, Stop 580000, Dir 0, Delay 0, Rtc 0.10 [s], Freq 0, PWM 0, Tilt 81
08:08:34.623 MQT: stat/tasmota_blinds_dnevna_desno/SHUTTER1 = 100
08:08:34.628 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":81}}
08:08:34.696 CFG: Saved to flash at F4, Count 316, Bytes 4096
08:08:35.612 ADE: ACCMODE 0x2D2800, VRMS 6329029, 6327673, Period 4476, 4476, IRMS 1773, 1773, WATT 0, 0, VA 28, 29, VAR 0, 0


08:08:35.922 CMD: Backlog ShutterPosition 100;ShutterTilt 90
08:08:35.924 SRC: WebConsole from 192.168.10.2
08:08:35.927 CMD: Grp 0, Cmd 'BACKLOG', Idx 1, Len 34, Pld -99, Data 'ShutterPosition 100;ShutterTilt 90'
08:08:35.974 SRC: Backlog
08:08:35.977 CMD: Grp 0, Cmd 'SHUTTERPOSITION', Idx 1, Len 3, Pld 100, Data '100'
08:08:35.981 SHT: Pos. in: payload 100 (3), payload 100, idx 1, src 9
08:08:35.983 SHT: lastsource 9:, real 580000, target 580000, tiltreal: 81, tilttarget: 0, payload 100
08:08:35.985 SHT: Delay Start? var1 <99>=<>, max10s?
08:08:35.987 SHT: VenetianDelay: 23, Pos: 81, Dir: -1, Delta: 90, Dur: 25, StartP: 81, TgtP: 0
08:08:35.989 SRC: Shutter
08:08:35.991 SHT: Switched relay 2 by Shutter
08:08:35.998 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"POWER2":"ON"}
08:08:36.002 MQT: stat/tasmota_blinds_dnevna_desno/POWER2 = ON
08:08:36.104 SRC: Shutter
08:08:36.107 SHT: Switched relay 1 by Shutter
08:08:36.116 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"POWER1":"ON"}
08:08:36.122 MQT: stat/tasmota_blinds_dnevna_desno/POWER1 = ON
08:08:36.128 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Position1":100}
08:08:36.133 SHT: Shtr1 Real 580000, Start 580000, Stop 580000, Dir -1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 81
08:08:36.138 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Shutter1":{"Position":100,"Direction":-1,"Target":100,"Tilt":81}}
08:08:36.141 SHT: Time 0(0), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 580000, nStop 0, Trgt 580000, mVelo 0, Dir -1, Tilt 81, TrgtTilt: 0, Tiltmove: 1
08:08:36.147 SHT: Shtr1 Real 580000, Start 580000, Stop 580000, Dir -1, Delay 0, Rtc 0.00 [s], Freq 0, PWM 0, Tilt 81
08:08:36.153 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Shutter1":{"Position":100,"Direction":-1,"Target":100,"Tilt":81}}
08:08:36.212 SRC: Backlog
08:08:36.215 CMD: Grp 0, Cmd 'SHUTTERTILT', Idx 1, Len 2, Pld 90, Data '90'
08:08:36.219 SHT: TiltTarget 90, payload -99
08:08:36.221 SHT: Pos. in: payload  (2), payload -99, idx 1, src 23
08:08:36.222 SHT: Stop 1 Mode 0 time 2
08:08:36.224 SRC: Shutter
08:08:36.226 SHT: Switched relay 1 by Shutter
08:08:36.232 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"POWER1":"OFF"}
08:08:36.237 MQT: stat/tasmota_blinds_dnevna_desno/POWER1 = OFF
08:08:36.240 SRC: Shutter
08:08:36.242 SHT: Switched relay 2 by Shutter
08:08:36.248 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"POWER2":"OFF"}
08:08:36.253 MQT: stat/tasmota_blinds_dnevna_desno/POWER2 = OFF
08:08:36.610 ADE: ACCMODE 0x2D2800, VRMS 6327277, 6327373, Period 4476, 4476, IRMS 1773, 1773, WATT 0, 70, VA 29, 132, VAR 0, 17
08:08:36.762 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Position1":100}
08:08:36.767 SHT: Time 2(1), cStop 0, cVelo 0, mVelo 0, aVelo 1000, mRun 0, aPos 0, aPos2 580000, nStop 0, Trgt 580000, mVelo 0, Dir 1, Tilt 78, TrgtTilt: 78, Tiltmove: 1
08:08:36.769 SHT: Stop 1 Mode 0 time 2
08:08:37.274 SHT: Shtr1 Real 580000, Start 580000, Stop 580000, Dir 0, Delay 0, Rtc 0.10 [s], Freq 0, PWM 0, Tilt 78
08:08:37.283 MQT: stat/tasmota_blinds_dnevna_desno/SHUTTER1 = 100
08:08:37.288 MQT: stat/tasmota_blinds_dnevna_desno/RESULT = {"Shutter1":{"Position":100,"Direction":0,"Target":100,"Tilt":78}}
```

It seems like final tilt actually never reaches 90? It reached 81 and 78 in above two commands.

**Reply by stefanbode** [reply]
People often expect that the tilt behaves like the position but this is not the truth. The granularity is much higher because changing the tilt is often just 1 second. This are 20 steps in TASMOTA. e.g. 0..90 = 4.5° per step. Additionally there is a minimum runtime required. You cannot change just 4.5°. The minimum is 3x 4.5°. This is all immutable because of the time slicing of TAMSOTA

**Comment by stefanbode** [comment]
Yes your finding is correct because the shutterposition 100 gets changed into -> shutterposition 100, shuttertilt 0 and this is from a tilt of 80 a doable amount of time therefore it starts. Then came in the shuttertilt 80 and it immediate stops. You will not have this effect in shutterposition 99 or something else because these positions do not touch the tilt. shutterposition 0 and 100 changes the tilt. Maybe a small modification may help.....

**Reply by stefanbode** [reply]
Change is on the way..... https://github.com/arendst/Tasmota/pull/17294

**Reply by matejdro** [reply]
Works perfectly, thanks a lot. You are a legend.
